composer.json

```diff
@@ -44,10 +44,10 @@
         "phpspec/prophecy-phpunit": "^2.0",
         "friendsofphp/php-cs-fixer": "^2.16",
         "phpstan/phpstan": "^0.12.2",
-        "sentry/sentry-symfony": "^3.4",
+        "sentry/sentry-symfony": "^4.0",
         "symfony/framework-bundle": "^4.0||^5.0",
         "nyholm/psr7": "^1.2",
-        "doctrine/mongodb-odm-bundle": "^4.1",
+        "doctrine/mongodb-odm": "dev-master#fa967eadf9e226923a97b145e8018f94d6580f09",
```

Comment:
2.2 seems to be released with PHP8 support.

Also can you add the `conflict` section?

Reply:
Ouch, this was alreadi fixed but I force-pushed :facepalm: 

I just realized I don't know how to do the conflict section, if we are targeting `sentry/sentry-symfony >= 4.0` which already targets `sentry/sentry => 3.0` I think it's impossible for this package to be installed with sentry 2.X. The problem is that upgrading the sentry-bundle and sentry will require manual intervention. Anyway, if they already changed to a major version, why should we do it? The users will have to edit their composer.json to allow upgrading it and thus they will expect things to require some manual changes.

Reply:
This package does not depend directy on `sentry/sentry-symfony` it's only in `require-dev` to test the integration.

You can already install `4.0` but the integration will not work find without your changes.

Maybe we can go for another solution and make the code compatible with sentry 2 & 3.

Maybe a static utility class can wraps `flush` & `setIpAdress` to do the right thing depending on the currently installed version. What do you think?

Reply:
ok, I did it. The most problematic part was providing different mocks for the interface `TransportFactoryInterface` in the `SentryMiddlewareTest`. Also I had to add some ignore rules for phpstan as it complains from the use of non-existing methods.

Reply:
phpstan will work fine if sentry 3.x is used, for sentry 2.x some rules should be added to `ignoreErrors`

Reply:
Cool :)

I sent a last batch of comments and we are good!

It's not a problem for PHPStan.